src/worktree.c

```diff
@@ -143,10 +143,14 @@ static int open_worktree_dir(git_worktree **out, const char *parent, const char
 		error = -1;
 		goto out;
 	}
-
-	if ((error = git_path_prettify_dir(&gitdir, dir, NULL)) < 0)
+	
```

Comment:
`git_path_dirname_r` is only necessary because the worktrees fs path should be the parent directory of the .git file within a worktree. It is unrelated to the /private/tmp thing. In the new branch I am creating I see that I could do it like this instead

```
	if ((wt->name = git__strdup(name)) == NULL
	    || (wt->commondir_path = git_worktree__read_link(dir, "commondir")) == NULL
	    || (wt->gitlink_path = git_worktree__read_link(dir, "gitdir")) == NULL
	    || (wt->parent_path = git__strdup(parent)) == NULL
	    || (wt->worktree_path = git_path_dirname(wt->gitlink_path)) == NULL) {
		error = -1;
		goto out;
	}
```

This seems a little cleaner to me and is clearer that it is unrelated to the other gitdir.
```

Reply:
Agreed.